### ISSUE TITLE
[FEATURE_REQUEST] New ObjectKind and Template for PersistentVolumeClaim and volumeClaimTemplates

### ISSUE THREAD

**Issue by hanife-Orthopy** [stackrox/kube-linter#815]
## Description of the problem/feature request
I would like to have checks available that allow me to check annotations on PersistentVolumeClaims (PVCs).
My main use-case is to check for annotations regarding data security and classification/compliance.
 
 
## Description of the existing behavior vs. expected behavior
Current behavior: I found no way to check annotations on PVCs or volumeClaimTemplates in StatefulSets.
Expected behavior: kube-linter should include the ability to check annotations on PVCs and volumeClaimTemplates
 
## Additional context:
 
To achieve this I think that the following things are necessary:
 
- A new `ObjectKind`: I would add a new ObjectKind for PersistentVolumeClaims. This can then be used to use existing checks like the required-annotation check.
- A new `Template`: To check for annotations on volume claim templates within StatefulSets. This template takes a parameter annotation and verifies its existence. The regular required-annotation check can not be used because, it would not correctly unwrap this object.
- A new `Extract`: To get the StatefulSetSpec extracted and use it to get at the volumeClaimTemplate.
 
I already have code that implements this and we are using it internally to check our manifests and would like to contribute this here as I think these checks would be beneficial to the community.
 
 
Thank you for considering these enhancements.
Hanife

**Comment by janisz** [comment]
@hanife-Orthopy That's great. I'd be happy to review and merge PVC support.

**Comment by hanife-Orthopy** [comment]
The related PR is https://github.com/stackrox/kube-linter/pull/922

**Comment by janisz** [comment]
@hanife-Orthopy - CEL expression solution using the new template:

- https://github.com/stackrox/kube-linter/pull/1012

## Simple Solution
```yaml
customChecks:
  - name: pvc-required-annotations
    template: cel-expression
    params:
      check: |
        object.kind == "PersistentVolumeClaim" && 
        (!has(object.metadata.annotations) || 
         !has(object.metadata.annotations["data.classification"])) ?
        "PVC must have data classification annotation" : ""
    scope:
      objectKinds: [PersistentVolumeClaim]
```
